src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -461,12 +461,12 @@ private static boolean skipDoubleBraceInstInit(Details details) {
     private static boolean isBlockAloneOnSingleLine(Details details) {
         DetailAST nextToken = details.nextToken;
 
-        while (nextToken != null && nextToken.getType() == TokenTypes.LITERAL_ELSE) {
-            nextToken = Details.getNextToken(nextToken);
+        if (nextToken != null && nextToken.getType() == TokenTypes.LITERAL_ELSE) {
+            nextToken = null;
```

Comment:
This is only for `ALONE_OR_SINGLELINE` and `TokenUtil.areOnSameLine(details.lcurly, details.rcurly)` must be `true` for this difference to matter.

If the token doesn't enter the loop, then it is never changed. So this literally only applies when the `nextToken` is an `LITERAL_ELSE`. So this looks like it comes down to the difference between `Details.getNextToken` and `null`.

It is actually all about `Details.getNextToken` and the loop conditions and the tree. `getNextToken` only looks at the next siblings or the next parent. `LITERAL_ELSE` has no siblings, so the parent only goes up the chain and grabs the next token after it, which can be whatever expression is next.

So what matters next is the condition on the return, `nextToken == null || !TokenUtil.areOnSameLine(details.rcurly, nextToken) || isRightcurlyFollowedBySemicolon(details)`. `nextToken == null` doesn't matter since that is what my mutated code is doing.

`!TokenUtil.areOnSameLine(details.rcurly, nextToken) || isRightcurlyFollowedBySemicolon(details)` is where I believe I found a test to kill this mutation.

Reply:
Should be killed now. Resolved.